contracts/core/Allo.sol

```diff
@@ -1,89 +1,289 @@
+    event FeeUpdated(uint256 fee);
+
+    /// ====================================
+    /// =========== Intializer =============
+    /// ====================================
+
+    /// @notice Initializes the contract after an upgrade
+    /// @dev During upgrade -> an higher version should be passed to reinitializer
+    /// @param _registry The address of the registry
+    function initialize(address _registry) public reinitializer(1) {
+        _initializeOwner(msg.sender);
+
+        registry = Registry(_registry);
+
+        __Multicall_init();
+    }
+
+    /// ====================================
+    /// =========== Modifier ===============
+    /// ====================================
 
-    // Public getter on the pools mapping
-    /// @notice calls out to the registry to get the identity metadata
-    function getPoolInfo(
-        uint256 _poolId
-    ) external view returns (PoolData memory, string memory) {
-        // Implement the function here
+    modifier isPoolMember(uint256 _poolId) {
+        registry.isMemberOfIdentity(pools[_poolId].identityId, msg.sender);
+        _;
     }
 
-    // @todo insert clonable strategy library, including validation that an existing pool has safe strategy
+    /// ====================================
+    /// ==== External/Public Functions =====
+    /// ====================================
+
+    /// @notice Fetch pool and identityMetadata
+    /// @param _poolId The id of the pool
+    /// @dev calls out to the registry to get the identity metadata
+    function getPoolInfo(uint256 _poolId) external view returns (Pool memory) {
+        return pools[_poolId];
+    }
 
     // creates pool locally, transfers pool amount to distribution strategy => returns poolId
     // takes fee from user
-    // validates that the owner is actually allowed to use the identity
+
+    /// @notice Creates a new pool
+    /// @param _identityId The identityId of the pool creator in the registry
+    /// @param _allocationStrategy The address of the allocation strategy
+    /// @param _distributionStrategy The address of the distribution strategy
+    /// @param _token The address of the token that the pool is denominated in
+    /// @param _amount The amount of the token to be deposited into the pool
+    /// @param _metadata The metadata of the pool
     function createPool(
-        PoolData memory /*_poolData*/
-    ) external pure returns (uint) {
-        uint32 _poolId = 0;
+        bytes32 _identityId,
+        address _allocationStrategy,
+        address payable _distributionStrategy,
+        address _token,
+        uint256 _amount,
+        Metadata memory _metadata
+    ) external payable returns (uint256 poolId) {
+        if (!registry.isMemberOfIdentity(_identityId, msg.sender)) {
+            revert NO_ACCESS_TO_ROLE();
+        }
+
+        Pool memory pool = Pool({
+            identityId: _identityId,
+            allocationStrategy: IAllocationStrategy(_createClone(_allocationStrategy)),
```

Comment:
move cloning logic into the new contract. 
createPool always uses the _allocationStrategy/ _distributionStrategy given to it

Reply:
@zobront @codenamejason @KurtMerbeth So i believe the direction we wanted to head was 
- Allo.sol would not invoke _createClone. Instead always use the address
- We would move clone logic into another library 

This is where I get a little fuzzy. The options discussed where: 
- We would have an additional contract `StrategyClone.sol` which exposes 1 function which would leverage Clone library to clone contract and then invoke Allo.createPool 
OR
- Where we have a whitelisted mapping of string(bytes) -> address in Allo. If you pass that string -> we'll do a look up and if we find a mapping -> clone , else not clone 
OR
- We just have createPoolWithClone and createPool in Allo.sol and leave it at that

Reply:
> Where we have a whitelisted mapping of string(bytes) -> address in Allo. If you pass that string -> we'll do a look up and if we find a mapping -> clone , else not clone

we should avoid going this path because it will be super confusing
I'm in favor of clone all, or if you decide against, we should have two functions where `createPool` checks if it's a whitelisted strategy or not. if yes -> `_createPoolWithClone`. Just to be safe

Reply:
Can we talk about edge cases of third party strategies?

**Scenario 1:**
In the first scenario, there are two pools:

In Pool 1, a third-party deploys a cloneable strategy that is not included in our whitelisted list. The pool is initialized using a clone of that strategy. 
In Pool 2, the pool is initialized using the implementation of the cloneable strategy directly, without making a clone.


**Scenario 2:**
In the second scenario, there are again two (or more) pools:
In Pool 1, a third-party strategy is deployed and the pool is initialized directly using that implementation without creating a clone. 
In Pool 2, the pool is also initialized with the same strategy, at the same address, again without creating.

This situation can potentially cause issues if the third-party strategy is poorly designed. Since both pools are using the same implementation at the same address, any problems or vulnerabilities in the strategy can affect both pools simultaneously.

Reply:
> I'm in favor of clone all, or if you decide against, we should have two functions where createPool checks if it's a whitelisted strategy or not. if yes -> _createPoolWithClone. Just to be safe

This makes sense. Easier to implement and our mapping adds a safety check
Personally the more I think about , the more it makes sense to always clone

Reply:
I am assuming we should have the whitelist of strategies onchain. I added a new contract called `Whitelist.sol` and a function in `Allo.sol` to call/check the addresses of the strategies the user is trying to deploy.

`Whitelist.sol` contains the following functions:
```javascript
mapping(address => bool) public whitelist;

constructor() {}

function addToWhitelist(address _address) external {
    whitelist[_address] = true;
}

function removeFromWhitelist(address _address) external {
    whitelist[_address] = false;
}

function isWhitelisted(address _address) external view returns (bool) {
    return whitelist[_address];
}
```

I didn't add any role management and they are wide open currently but didn't want to go all in if this was not a path we wanted to pursue.

Reply:
Open question :
- should this live within allo.sol
- have 2 functions createPool and createPoolWithClone 

@zobront

Reply:
We've rewritten this to defer clonability to the strategy 
aa93a6d0b087741c0c142293bc4c815a23039237 

Curious what you think
@zobront + @0xZakk

Reply:
I am in favor of it being a separate contract but it doesn't have to be. The important thing is that it's separated into two functions.

`createPool()` => takes in addresses for strategies and uses them, no clones
`createPoolWithClone()` => uses takes in strategies from approved list, clones them, then calls `createPool()` using those clones as arguments, and marks it as "approved" because strategies came from library

You know I like consolidating into fewer contracts but I am in favor of these being separate because it keeps the logic of Allo.sol clean (it uses the addresses its passed as strategies) and makes the StrategyLibrary.sol a peripheral contract that Allo owns, can be easily updated, can track all the approved strategies, etc. 

But your call, this is fine either way.